client/app/components/Header/index.jsx

```diff
@@ -18,23 +18,61 @@ export type Props = {
 
 export type State = {
   mobileNavOpen: boolean,
-  toggled: boolean,
 };
 
 export const Header = ({
   home, links, mobileOnly, profile,
 }: Props): Node => {
   const [mobileNavOpen, setMobileNavOpen] = useState(false);
-  const [toggled, setToggled] = useState(true);
+  const navigationRef = useRef(null);
+
+  const handleNavigationKeyDown = (
+    event: SyntheticKeyboardEvent<HTMLElement>,
+  ): void => {
+    if (mobileNavOpen) {
+      // Let's trap the focus
+      if (navigationRef.current) {
+        const focusableElements = navigationRef.current.querySelectorAll(
+          'a, button, input, select, textarea, [tabindex]:not([tabindex="-1"])',
+        );
+        const firstFocusableElement = focusableElements[0];
+        const lastFocusableElement = focusableElements[focusableElements.length - 1];
+
+        if (event.key === 'Tab') {
```

Comment:
When I do the following, the focus trap does not work:

1. `Tab` into the hamburger icon
2. Hit `Enter` to open the mobile menu
3. `Tab` through the menu items
4. When the last item of the menu is reached, and I hit `Tab` again, the focus trap does not work and the contents behind the menu are hit.

The expected behaviour is for the focus to go back to logo at the top of the menu.

![Oct-11-2022 21-21-15](https://user-images.githubusercontent.com/3010728/195250484-c39aa219-6ef8-4405-bc66-7320f35c3f97.gif)

Reply:
I think the problem is that `document.activeElement` is not giving you the active element that is being focused on. It's giving the element before.

I think we want to make t[his focus trap code we're using in the Modal component](https://github.com/ifmeorg/ifme/blob/main/client/app/components/Modal/index.jsx#L64) a custom hook that we can reuse here.

Reply:
Another issue I see is that `lastFocusableElement` is the "Sign out" button when it should be the "Resources" link.

Reply:
@julianguyen Very interesting! I was testing it in an unauthenticated state on the homepage–I feel like this is a separate state. I'll take a look (and look at lifting the `Modal` component `useEffect` into a reusable hook! 💪🏻 )

Reply:
This is fixed now! It was related to a `display: none` element in the Links - it isn't technically focusable by the browser but still appeared in our `focusableElements` array. I've added an extra step to filter out non-visible elements here – the Focus Trap now works as expected while authenticated or unauthenticated.